llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp

```diff
@@ -2157,6 +2221,16 @@ bool SPIRVInstructionSelector::selectIntrinsic(Register ResVReg,
     break;
   case Intrinsic::spv_thread_id:
     return selectSpvThreadId(ResVReg, ResType, I);
+  case Intrinsic::spv_fdot:
+    return BuildMI(BB, I, I.getDebugLoc(), TII.get(SPIRV::OpDot))
```

Comment:
Maybe we should do a `selectDot` function? Would be more consistent with how all the other cases are setup. Also you are doing operand checks before addUse in `selectIntegerDot` that you aren't doing here.

Reply:
I don't mind creating a float function to expand the fdot. I don't think you're suggesting that I have one call for float and integer as that's the opposite of what you suggested [here](https://github.com/llvm/llvm-project/pull/102872#discussion_r1714053193). Are the checks you're referring to the asserts?

Reply:
no i'm saying create a function so that its cleaner to asserts on the number of operands and the validity of those operands.

Reply:
Done assuming I've understood the suggestion correctly.